libraries/chain/transaction_context.cpp

```diff
@@ -85,6 +85,28 @@ namespace bacc = boost::accumulators;
    volatile sig_atomic_t deadline_timer_verify::hit;
    static deadline_timer_verify deadline_timer_verification;
 
+   void provided_bandwith::confirm(account_name provider) {
+       verify_limits_not_confirmed();
+       confirmed_ = true;
+       provider_ = provider;
+   }
+
+   void provided_bandwith::set_net_limit(int64_t net_limit) {
+       verify_limits_not_confirmed();
+       this->net_limit_ = net_limit;
+   }
+
+   void provided_bandwith::set_cpu_limit(int64_t cpu_limit) {
+       verify_limits_not_confirmed();
+       this->cpu_limit_ = cpu_limit;
+   }
+
+   void provided_bandwith::verify_limits_not_confirmed() {
+       if (confirmed_) {
+           EOS_THROW( bandwith_confirmed, "bandwith has been already confirmed. No changes could be done");
```

Comment:
May be `bandwidth_already_confirmed`?

Reply:
`EOS_ASSERT(confirmed_, bandwith_already_confirmed, '')` - looks more simple for this case.

Reply:
!confirmed_